ocs_ci/helpers/cephx_keyrotation_helper.py

```
@@ -0,0 +1,4406 @@
+        phase = (storage_cluster.data.get("status") or {}).get("phase")
+        message = (storage_cluster.data.get("status") or {}).get("message", "")
+        raise UnexpectedBehaviour(
+            "StorageCluster did not reach Ready within "
+            f"{timeout}s after CephX key rotation "
+            f"(last phase={phase}, message={message})"
+        )
+
+    def wait_for_rook_daemon_pods_ready(self, timeout=600):
+        """Wait until MON, MGR, OSD, and MDS pods are Running."""
+        for daemon, label in constants.ROOK_CEPHX_KEYROTATION_DAEMON_LABELS.items():
+            log.info(f"Waiting for {daemon} pods ({label}) to be Running")
```

Ruff (0.15.21)

[warning] Logging statement uses f-string

(G004)